src/Foundatio/Lock/ThrottlingLockProvider.cs

```diff
@@ -148,3 +148,31 @@ private string GetCacheKey(string resource, DateTime now)
         return String.Concat(resource, ":", now.Floor(_throttlingPeriod).Ticks);
     }
 }
+
+public interface IThrottlingLockProviderFactory
+{
+    ILockProvider Create(int maxHitsPerPeriod = 100, TimeSpan? throttlingPeriod = null);
+}
+
+public class ThrottlingLockProviderFactory : IThrottlingLockProviderFactory
+{
+    private readonly ICacheClient _cacheClient;
+    private readonly TimeProvider _timeProvider;
+    private readonly IResiliencePolicyProvider _resiliencePolicyProvider;
+    private readonly ILoggerFactory _loggerFactory;
+
+    public ThrottlingLockProviderFactory(ICacheClient cacheClient, TimeProvider timeProvider = null,
+        IResiliencePolicyProvider resiliencePolicyProvider = null, ILoggerFactory loggerFactory = null)
+    {
+        _cacheClient = cacheClient ?? throw new ArgumentNullException(nameof(cacheClient));
+        _timeProvider = timeProvider ?? cacheClient.GetTimeProvider() ?? TimeProvider.System;
+        _resiliencePolicyProvider = resiliencePolicyProvider ?? cacheClient.GetResiliencePolicyProvider();
```

Comment:
The more time that goes on, the more this feels really odd to pull these things from different building blocks and then guessing where you got it from.